src/engine/strat_engine/engine.rs

```diff
@@ -124,6 +140,49 @@ impl Engine for StratEngine {
         Ok(uuid)
     }
 
+    /// Evaluate a device node & devicemapper::Device to see if it's a valid
+    /// stratis device.  If all the devices are present in the pool and the pool isn't already
+    /// up and running, it will get setup and the pool uuid will be returned.
+    fn block_evaluate(&mut self, dev_node: PathBuf, device: Device) -> Option<PoolUuid> {
+        let mut duplicate = false;
+
+        match is_stratis_device(&mut self.incomplete_pools,
+                                dev_node,
+                                libc::dev_t::from(device)) {
+            Ok(process_it) => {
+                let mut rc = None;
+                if let Some(pool_uuid) = process_it {
+                    if let Some(devices) = self.incomplete_pools.get(&pool_uuid) {
```

Comment:
I think that ```self.incomplete_pools.get(&pool_uuid)``` will always return ```Some``` value. The reason for that is that ```is_stratis_device``` always inserts into the hashmap that is its first argument whenever it returns a uuid. This appears to me to be a definite bug.

This goes back to my other point, which is that ```is_stratis_device``` would be a lot handier if did not take the hashmap argument but left it to the user to decide what values to be inserting where.